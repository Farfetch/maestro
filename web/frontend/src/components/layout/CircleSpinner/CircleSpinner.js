import { LoadingOutlined } from "@ant-design/icons";
import { Spin } from "antd";

const CircleSpinner = ({ size = 32 }) => (
  <Spin indicator={<LoadingOutlined style={{ fontSize: size }} spin />} />
);

export default CircleSpinner;
