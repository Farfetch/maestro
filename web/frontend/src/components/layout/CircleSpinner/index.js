import CircleSpinner from "./CircleSpinner";

export default CircleSpinner;
