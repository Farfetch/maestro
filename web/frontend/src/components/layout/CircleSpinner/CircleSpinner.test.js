import { render } from "@testing-library/react";

import CircleSpinner from "./CircleSpinner";

describe("components/form/select/CircleSpinner", () => {
  test(`should render CircleSpinner component`, async () => {
    const rendered = render(<CircleSpinner />);

    const badgeComponent = rendered.container;
    expect(badgeComponent.outerHTML).toMatchSnapshot();
  });
});
