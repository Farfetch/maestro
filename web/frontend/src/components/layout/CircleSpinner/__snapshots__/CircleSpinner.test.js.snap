// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/form/select/CircleSpinner should render CircleSpinner component 1`] = `"<div><div class=\\"ant-spin ant-spin-spinning css-dev-only-do-not-override-nnuwmp\\" aria-live=\\"polite\\" aria-busy=\\"true\\"><span role=\\"img\\" aria-label=\\"loading\\" style=\\"font-size: 32px;\\" class=\\"anticon anticon-loading anticon-spin ant-spin-dot\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"loading\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z\\"></path></svg></span></div></div>"`;
